pythran/transformations/expand_globals.py

```diff
@@ -0,0 +1,82 @@
+"""
+ExpandGlobals replaces globals variables by function call.
+
+It also turn globals assignment in function definition.
+"""
+
+from pythran.analyses import LocalDecl
+from pythran.passmanager import Transformation
+from pythran import metadata
+
+import ast
+
+
+class ExpandGlobals(Transformation):
+
+    """
+    Expands all builtins into full paths.
+
+    >>> import ast
+    >>> from pythran import passmanager, backend
+    >>> node = ast.parse('''
+    ... a = 1
+    ... def foo():
+    ...     return a''')
+    >>> pm = passmanager.PassManager("test")
+    >>> _, node = pm.apply(ExpandGlobals, node)
+    >>> print pm.dump(backend.Python, node)
+    def a():
+        return 1
+    def foo():
+        return a()
+    """
+
+    def __init__(self):
+        """ Initialize local declaration and constant name to expand. """
+        self.local_decl = set()
+        self.to_expand = set()
+        super(ExpandGlobals, self).__init__()
+
+    def visit_Module(self, node):
+        """Turn globals assignement to functionDef and visit function defs. """
+        module_body = list()
+        # Gather top level assigned variables.
+        for stmt in node.body:
+            if not isinstance(stmt, ast.Assign):
+                continue
+            for target in stmt.targets:
+                assert isinstance(target, ast.Name)
```

Comment:
Not only, I should filter, import and functions (indeed, everything except assignement to ast.Name)

I agree it looks like an analyse. I was just not sure we should create an analyse for this "very simple" case.